### PR TITLE
Fix public docs for chbench business intelligence demo

### PR DESCRIPTION
**Pull request by quodlibetor**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3025)

### CODE REVIEW

File: doc/user/content/demos/business-intelligence.md [hunk 1]
```diff
@@ -118,29 +118,18 @@ Metabase](/images/demos/bi_architecture_diagram.png)
...
+    quite a bit of time (up to 10 minutes, even on very fast connections).
 
     ```shell
-    # Deploy MySQL, Debezium, Kafka et. al., Metabase, and Materialize.
-    # Initialize MySQL with some chBench data.
-    ./dc.sh up :init:
-    ./dc.sh up :demo:
-    # Run the chBench workload generator.
-    ./dc.sh demo-load
+    bin/mzconduct run chbench --workflow demo
```
Comment (cuongdo) [inline doc/user/content/demos/business-intelligence.md:127]: The command line doesn't work for me with the latest code from `master`:

``` bash-session
[cdo@ubuntu:materialize]$ bin/mzconduct run chbench --workflow demo
C> Executing chbench -> demo
C> ERROR: Unknown workflow: 'demo'. Expected one of: ci, demo-load, load-test, bring-up-source-data
```

Looks like it's `demo-load`?

Comment (quodlibetor) [reply doc/user/content/demos/business-intelligence.md:127]: This PR also changes the name of that workflow to demo.

Comment (quodlibetor) [reply doc/user/content/demos/business-intelligence.md:127]: I've added back a `demo-load` workflow that just delegates to the `demo` workflow, I'm not sure if we should prefer to not update the docs page to the new command until the next release of materialize for some reason @sploiselle @benesch?

Comment (benesch) [reply doc/user/content/demos/business-intelligence.md:127]: Yeah, we don't have a good story about backwards incompatible changes like this at the moment. I don't really know what to tell you! Both options are bad! If you wait, you'll probably forget. (At least if I were in your shoes I would.) If you do it now, the docs will be a bit wrong until the next release. Maybe @sploiselle has a magic bullet.

Comment (quodlibetor) [reply doc/user/content/demos/business-intelligence.md:127]: ah so maybe leave the docs using the old thing, and then eventually point to the new thing, when we remember. To be clear it's not a big deal for them to have both, I just think `demo-load` is kind of the wrong word.

Comment (quodlibetor) [reply doc/user/content/demos/business-intelligence.md]: Which I have now done.

**Comment by quodlibetor** [comment]
I'm going to merge this @cuongdo but I am happy to make further changes if it's currently confusing, since what's currently up is just wrong.

**Pull request merged**